### PR TITLE
Send events when opening and finalizing events extraction as function

### PR DESCRIPTION
**Pull request by AlexandreSi**
So that we know how often this feature is used, and if we modify/market it, we can track how good/bad it works

### CODE REVIEW

File: newIDE/app/src/EventsSheet/index.js [hunk 1]
```diff
@@ -1502,6 +1504,7 @@ export class EventsSheetComponentWithoutHandle extends React.Component<
                         serializedEventsToExtract: null,
                       })
                     }
+                    onOpen={onBeginCreateEventsFunction}
```
Comment (4ian) [inline newIDE/app/src/EventsSheet/index.js:1507]: I would avoid `onOpen` (the parent is the one responsible for the child, so they should know when they open it) and add the call to onBeginCreateEventsFunction in `extractEventsToFunction`

**Review by 4ian — APPROVED** [review]
LGTM without the onOpen

**Comment by D8H** [comment]
I wonder if data about cancelled forms could be even more infighting than than filled forms.

**Comment by AlexandreSi** [comment]
> I wonder if data about cancelled forms could be even more in-lighting than than filled forms.
> 
> * Do people see the form and get scared away or do they try to fill something and get stuck?

That's a piece of information that we should be able to deduct from this: Out of 53 openings, only 3 went through => 50 cancellations. Or do you feel we will miss something with this logic?

File: newIDE/app/src/EventsSheet/index.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/EventsSheet/index.js:1507]: True, I will change that

**Comment by D8H** [comment]
> > I wonder if data about cancelled forms could be even more in-lighting than than filled forms.
> > 
> > * Do people see the form and get scared away or do they try to fill something and get stuck?
> 
> That's a piece of information that we should be able to deduct from this: Out of 53 openings, only 3 went through => 50 cancellations. Or do you feel we will miss something with this logic?

What is filled when the dialog is cancelled could help understand what was not understood, but maybe this is too much data to send or not really necessary. I'm not sure.

**Comment by AlexandreSi** [comment]
> What is filled when the dialog is cancelled could help understand what was not understood, but maybe this is too much data to send or not really necessary. I'm not sure.

I think that's too much data for the service we use (hard to visualize).

Identify what is blocking users on this dialog could be the goal of a UX study.

**Pull request merged**